recipe-server/client/control_new/components/common/BaseAntForm.js

```diff
@@ -0,0 +1,94 @@
+import autobind from 'autobind-decorator';
+import { Form, message } from 'antd';
+import React, { PropTypes as pt } from 'react';
+
+
+/**
+ * Base class for antd-based forms that handles displaying errors
+ * from the API.
+ */
+export default class BaseAntForm extends React.Component {
+  static propTypes = {
+    form: pt.object.isRequired,
+    onSubmit: pt.func.isRequired,
+    errors: pt.object,
+  }
+
+  @autobind
```

Comment:
I didn't know if using the autobind decorator on a base class would autobind methods on a class that inherits from it, and to which instance.

I just tested it, and it seems to only autobind the inherited method; methods that aren't on the base class, or overridden methods, do not appear to be autobound, so I'll fix this.